Makefile

```
@@ -250,20 +254,29 @@ vet: ## Run go vet against code.
 	go vet ./...
 
 .PHONY: clean-cov
-clean-cov: ## Remove coverage report
-	rm -rf cover.out
+clean-cov: ## Remove coverage reports
+	rm -rf coverage
 
 .PHONY: test
 test: test-unit test-integration ## Run all tests
 
 test-integration: clean-cov generate fmt vet envtest ## Run Integration tests.
-	KUBEBUILDER_ASSETS="$(shell $(ENVTEST) $(ARCH_PARAM) use $(ENVTEST_K8S_VERSION) -p path)" USE_EXISTING_CLUSTER=true go test ./... -coverprofile $(PROJECT_PATH)/cover.out -tags integration -ginkgo.v -ginkgo.progress -v -timeout 0
+	@export KUBEBUILDER_ASSETS="$(shell $(ENVTEST) $(ARCH_PARAM) use $(ENVTEST_K8S_VERSION) -p path)"; export USE_EXISTING_CLUSTER=true; \
+	for dir in $(INTEGRATION_DIRS); do \
+		echo "Running integration tests in $$dir..."; \
+		mkdir -p coverage/integration/$$dir; \
+		go test ./$$dir -coverprofile $(PROJECT_PATH)/coverage/integration/$$dir/cover.out -tags integration -ginkgo.v -ginkgo.progress -v -timeout 0; \
+		done
 
 ifdef TEST_NAME
 test-unit: TEST_PATTERN := --run $(TEST_NAME)
 endif
 test-unit: clean-cov generate fmt vet ## Run Unit tests.
-	go test ./... -coverprofile $(PROJECT_PATH)/cover.out -tags unit -v -timeout 0 $(TEST_PATTERN)
+	@for dir in $(UNIT_DIRS); do \
+		echo "Running unit tests in $$dir..."; \
+		mkdir -p coverage/unit/$$dir; \
+		go test ./$$dir -coverprofile $(PROJECT_PATH)/coverage/unit/$$dir/cover.out -tags unit -v -timeout 0 $(TEST_PATTERN); \
+	done
```

```suggestion
	go test $(UNIT_DIRS) -coverprofile $(PROJECT_PATH)/cover.out -tags unit -v -timeout 0 $(TEST_PATTERN)
```
May be able to do something like this if we choose to use the recursive directory scanning. One possible disadvantage to individually running them may be that a failed test may not be entirely obvious since it's not run as part of a single suite 🤔